src/main/java/Duke.java

```diff
@@ -1,10 +1,170 @@
+    });
+
+    userInput.setOnAction((event) -> {
+        handleUserInput();
+    });
+
+        dialogContainer.heightProperty().addListener((observable) -> scrollPane.setVvalue(1.0));
+    }
+
+    /**
+     * Iteration 1:
+     * Creates a label with the specified text and adds it to the dialog container.
+     * @param text String containing text to add
+     * @return a label with the specified text that has word wrap enabled.
+     */
+    private Label getDialogLabel(String text) {
+        // You will need to import `javafx.scene.control.Label`.
+        Label textToAdd = new Label(text);
+        textToAdd.setWrapText(true);
+
+        return textToAdd;
+    }
+
+    /**
+     * Iteration 2:
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    private void handleUserInput() {
+        Label userText = new Label(userInput.getText());
+        Label dukeText = new Label(getResponse(userInput.getText()));
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(userText, new ImageView(user)),
+                DialogBox.getDukeDialog(dukeText, new ImageView(duke))
+        );
+        userInput.clear();
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+
+    private String getResponse(String input) {
+        try {
+            if (input.equals("bye")) {
+                return ui.exit();
+            } else if (input.equals("list")) {
+                return ui.list();
+            } else if (input.startsWith("done")) {
+                int n = parser.getTaskNumber(input);
+                Task t = tasklist.complete(n);
+                return ui.complete(t);
+            } else if (input.startsWith("delete")) {
+                int n = parser.getTaskNumber(input);
+                Task t = tasklist.delete(n);
+                return ui.delete(t, tasklist.getTotal());
+            } else if (input.startsWith("todo") || input.startsWith("deadline") || input.startsWith("event")) {
+                Task t = tasklist.add(input);
+                return ui.add(t, tasklist.getTotal());
+            } else if (input.startsWith("find")) {
+                String keyword = parser.getKeyword(input);
+                return ui.find(keyword);
+            } else { // unknown input
+                throw new UnknownInputException();
+            }
+        } catch (DukeException | IOException e) {
+            return ui.handleException(e);
+        }
+    }
+
+
+    static void initialize() throws IOException {
+        Scanner myReader = new Scanner(PATH);
+        int total = myReader.nextInt();
+        tasklist = new TaskList(total);
+    }
+
+    public static void main(String[] args) throws IOException {
+        // use launcher to launch
     }
```

Comment:
Maybe this should be removed? Or possibly throw an error/exception rather than a comment.